### ISSUE TITLE
feat: `wrangler dev` should use `migrations`

### ISSUE THREAD

**Issue by threepointone** [cloudflare/wrangler2#736]
Right now we don't send migrations to `wrangler dev` because it doesn't support it. Without it, we won't be able to do `renamed_classes` during development, and the edge will simply consider `to` to be a new class. 

cc @a-robinson

**Comment by a-robinson** [comment]
So I've double-checked the source code for the API service, and it's already supposed to already work. Have you actually tried it and seen failures or was it just assumed that it didn't work?

To be clear, assuming that it wasn't supported was reasonable; but it is supposed to work.

**Comment by threepointone** [comment]
Might just be a miss on our part here then, I think it maybe used to fail a while ago? I'll try it tonight and update this thread. Thank you for checking!

**Comment by a-robinson** [comment]
Any update on this? I want to make sure I don't miss it from the list I'm compiling if this does still need any work.

**Comment by threepointone** [comment]
I've struggled to write any code at all this week 😭 Top of my list of things to hack on, might just start today. Sorry!

**Comment by a-robinson** [comment]
You certainly aren't alone there :) No worries!

**Comment by a-robinson** [comment]
Thanks, of course, that makes sense.

**Comment by penalosa** [comment]
@threepointone @a-robinson is this still an issue?

**Comment by a-robinson** [comment]
@penalosa yeah, it doesn't look like there's been any movement on it since our comments on #791

**Comment by JacobMGEvans** [comment]
@a-robinson Wanted to check on the status of this and the block? 

Any additional details or timeline?

**Comment by a-robinson** [comment]
Doesn't look like any changes have been made to the internal code for handling preview uploads to let them support DO migrations. If this is important, I'd get it onto the roadmap of either @bcaimano or @elithrar

**Comment by penalosa** [comment]
This is a very old issue so I'm going to close it for now. Today, we strongly recommend using local mode development (the default for `wrangler dev`), with [remote bindings](https://developers.cloudflare.com/changelog/2025-06-25-getplatformproxy-support-remote-bindings/) for accessing remote resources during a dev session.
